st_preview/preview_math.py

```diff
@@ -143,15 +144,51 @@ def _create_image(latex_program, latex_document, base_name, color,
         ])
 
     err_file_path = image_path + _ERROR_EXTENSION
+    err_log = []
     if not pdf_exists:
-        with open(err_file_path, "w") as f:
-            f.write(
-                "Failed to run '{latex_program}' to create pdf to preview."
-                .format(**locals())
-            )
+        err_log.append(
+            "Failed to run '{latex_program}' to create pdf to preview."
+            .format(**locals())
+        )
+        err_log.append("")
+        err_log.append("")
+        err_log.append("LaTeX document:")
+        err_log.append("-----BEGIN DOCUMENT-----")
+        err_log.append(latex_document)
+        err_log.append("-----END DOCUMENT-----")
+
+        err_log.append("")
+
+        log_file = os.path.join(temp_path, base_name + ".log")
```

Comment:
Might it be better to show the (parsed) errors before the document? Assuming the parsing code mostly does its job, this would mean that the file starts with what is probably the most relevant information.